File: benchmarks/vm/14_short_circuit_loop.py
import time

t0 = time.monotonic()
hits = 0
skips = 0
i = 0
while i < 200_000:
    d = i % 11
    if d != 0 and (100 // d) > 9:
        hits += 1
    if d == 0 or (100 // d) < 4:
        skips += 1
    i += 1
score = (hits * 3) + (skips * 7)
total_ms = (time.monotonic() - t0) * 1000
print(f"hits = {hits}")
print(f"skips = {skips}")
print(f"score = {score}")
print(f"BENCH_RESULT: {total_ms:.3f} ms")

File: benchmarks/vm/14_short_circuit_loop.lua
local t0 = os.clock()
local hits = 0
local skips = 0
local i = 0
while i < 200000 do
    local d = i % 11
    if d ~= 0 and (100 // d) > 9 then
        hits = hits + 1
    end
    if d == 0 or (100 // d) < 4 then
        skips = skips + 1
    end
    i = i + 1
end
local score = (hits * 3) + (skips * 7)
local total_ms = (os.clock() - t0) * 1000
print("hits = " .. tostring(hits))
print("skips = " .. tostring(skips))
print("score = " .. tostring(score))
print(string.format("BENCH_RESULT: %.3f ms", total_ms))
